login/core/validations/SignInValidation.js

```diff
@@ -0,0 +1,14 @@
+const mailValidation = require('./MailValidation.js');
+const passValidation = require('./PasswordValidation.js');
+const loginValidation = require('./LoginValidation.js');
+const CreateUser = require('../../db/UserResponseServices.js');
+
+
+function SignInValidation(loginRequest, response) {
+  if ((mailValidation.MailValidation(loginRequest.login) || loginValidation.LoginValidation(loginRequest.login)) && passValidation.PasswordValidation(loginRequest.password)) {
```

Comment:
I'd like to have this validation as middleware or an interceptor(rather the former) so it runs automatically on every request and you don't have to put it inside the controller method, because it breaks Single Responsibility Principle.

Reply:
May be i dont understand your case,but here https://jasonwatmore.com/post/2018/06/14/nodejs-mongodb-simple-api-for-authentication-registration-and-user-management and even here https://code.tutsplus.com/ru/tutorials/site-authentication-in-nodejs-user-sign-up--cms-29933
validation at the same route as a registration